lti_consumer/api.py

```diff
@@ -0,0 +1,44 @@
+"""
+Python APIs used to handle LTI configuration and launches.
+
+Some methods are meant to be used inside the XBlock, so they
+return plaintext to allow easy testing/mocking.
+"""
+from .models import LtiConfiguration
+
+
+def get_or_create_local_lti_config(lti_version, block_location):
+    """
+    Retrieves the id of the LTI Configuration for the
+    block and location, or creates one if it doesn't exist.
+
+    Doesn't take into account the LTI version of the cofiguration,
+    and updates it accordingly.
+
+    Returns LTI configuration ID.
+    """
+    lti_config, _ = LtiConfiguration.objects.get_or_create(
+        location=block_location,
+        config_store=LtiConfiguration.CONFIG_ON_XBLOCK,
+    )
+
+    if lti_config.version != lti_version:
+        lti_config.version = lti_version
```

Comment:
This means configs are overwritten when changing versions, right? Would it make sense to keep both?

Reply:
@viadanna Yes, but this will only happen for configuration stored in the block.
Also, since the configuration is still stored in block variables, you a LTI 1.1 block is configured and then switched to LTI 1.3, the settings will be kept.

This overriding just happen for configurations stored with the `CONFIG_ON_XBLOCK` type.